packages/twenty-server/src/workspace/messaging/utils/filter-out-participants-from-company-or-workspace.util.ts

```diff
@@ -0,0 +1,20 @@
+import { Participant } from 'src/workspace/messaging/types/gmail-message';
+import { getDomainNameFromHandle } from 'src/workspace/messaging/utils/get-domain-name-from-handle.util';
+import { WorkspaceMemberObjectMetadata } from 'src/workspace/workspace-sync-metadata/standard-objects/workspace-member.object-metadata';
+import { ObjectRecord } from 'src/workspace/workspace-sync-metadata/types/object-record';
+
+export function filterOutParticipantsFromCompanyOrWorkspace(
+  participants: Participant[],
+  selfHandle: string,
+  workspaceMembers: ObjectRecord<WorkspaceMemberObjectMetadata>[],
+): Participant[] {
+  const selfDomainName = getDomainNameFromHandle(selfHandle);
+
+  return participants.filter(
+    (participant) =>
+      getDomainNameFromHandle(participant.handle) !== selfDomainName &&
+      !workspaceMembers.some(
```

Comment:
nit: You can improve this part with a hashmap / groupBy instead of looping again here